### PR TITLE
NN-843

### PR DESCRIPTION
**Pull request by dzinad**

### CODE REVIEW

**Review by DzmitryFomchyn — APPROVED** [review]

File: changelog/unreleased/issues/changes.md [hunk 1]
```diff
@@ -0,0 +1 @@
+- In case an app provides custom `Router` implementation to Nav SDK, the Nav SDK doesn't trigger `NavigationRouteAlternativesObserver` with online alternatives when current route is offline.
```
Comment (dzinad) [inline changelog/unreleased/issues/changes.md:1]: Is it really so? I thought it was more like "it's triggered, but much later". The alternatives mechanism is still working, right? we just won't trigger an alternative request as soon as the network is back.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 2]
```diff
@@ -471,7 +471,11 @@ class MapboxNavigation @VisibleForTesting internal constructor(
             ),
             navigationOptions.accessToken ?: "",
             if (moduleRouter.isInternalImplementation()) {
-                nativeRouter
+                // We pass null to let NN know that default router is used and it can rely
+                // on implementation details in some cases like offline-online switch.
+                // Meanwhile platform SDK uses its own instance of native router for
+                // route requests
+                null
```
Comment (dzinad) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:478]: We also pass nativeRouter in `recreate` method. It should also be null.

File: changelog/unreleased/issues/changes.md (hunk 1 above)
Comment (VysotskiVadim) [reply changelog/unreleased/issues/changes.md:1]: @averkhaturau, can you please clarify consequences of using custom router for offline-online switch?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 2 above)
Comment (VysotskiVadim) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:478]: nice one! fixing in a followup PR

Comment (VysotskiVadim) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:478]: https://github.com/mapbox/mapbox-navigation-android/pull/7357

File: changelog/unreleased/issues/changes.md (hunk 1 above)
Comment (averkhaturau) [reply changelog/unreleased/issues/changes.md:1]: NN knows which router is online only if it creates the routers itself. Providing a custom router makes this information unavailable, we don't know how to create an online router anymore. That's why the back online feature will not be available with a custom router

**Pull request merged**